straxbra/plugins.py

```diff
@@ -368,58 +372,80 @@ def compute(self, peaks):
+    https://arxiv.org/abs/1112.1481 ).
+    '''
     __version__ = "0.0.1"
     dtype = [('x', np.float32,
-              'Reconstructed S2 X position (cm), uncorrected'),
+              'Reconstructed S2 X position (mm), uncorrected'),
              ('y', np.float32,
-              'Reconstructed S2 Y position (cm), uncorrected')]
+              'Reconstructed S2 Y position (mm), uncorrected')]
     depends_on = ('peaks',)
 
     parallel = False
 
-    pmt_locations = np.zeros((7,2))  # top only
-
     daq_to_location_map = np.arange(7)  # s2-top-channel to pmt-location
 
-    def setup(self):
-
-        self.pmt_mask = self.config['to_pe'][:self.config['top_pmts']] > 0
-        pmt_angles = np.linspace(0, 2*np.pi, 6, endpoint=False)
-        angle_offset = -2*np.pi/3  # position of PMT1
-        pmt_angles = pmt_offset - pmt_angles
-        pmt_radius = 30  # mm
-        self.pmt_positions = np.column((np.cos(pmt_angles),
-                                        np.sin(pmt_angles)))* pmt_radius
-        self.pmt_positions = np.append(self.pmt_positions, [[0,0]])
+    ## Function to return non-negative value corresponding to (radial position - radius TPC) if inside TPC,
+    ## used for constraints
+    @staticmethod
+    def insidevolume(inputs):
+        p_TPC_radius = 35
+        return (p_TPC_radius - np.sqrt(inputs[0]**2 + inputs[1]**2))
 
+    ## Radial LRF model
+    ## from: Position Reconstruction in a Dual Phase Xenon Scintillation Detector (https://arxiv.org/abs/1112.1481)
+    @staticmethod
+    def eta(r, A, r0, a, b, alpha):
+        return A * exp( - a * (r/r0) / (1 + (r/r0) ** (1 - alpha)) - b / (1 + (r/r0) ** (- alpha)))
+
+    ## Position dependent LRF values individual PMTs from model
+    def LRF_PMTs(x, y, self):
+        ## PMT positions
+        PMT_position_x = np.array([-14.,-28,-14.,14.,28.,14.,0.])      # x-position PMTs in mm in cartesian coordinates
+        PMT_position_y = np.array([-28.,0.,28.,28.,0.,-28.,0.])        # y-position PMTs in mm in cartesian coordinates
+        ## Fit parameters LRFs
+        ## MC driven (R_PTFE = 95 %, T_meshes = 89.770509 %, lambda_LXe = 100 cm);
+        ## To be iteratively determined from data later
+        fitparameters = np.array([np.array([ 0.58534179, 29.89341846, -0.3275816, 4.14715081, 2.61234684]), np.array([ 0.63546614, 28.49525342, -0.19651583, 3.76011493, 2.68672152]), np.array([ 0.58840586, 30.39015033, -0.38248759, 4.27642323, 2.56774842]), np.array([ 0.59111988, 31.38350968, -0.5760228, 4.64122418, 2.51311592]), np.array([ 0.63771524, 28.90663204, -0.26194541, 3.90052756, 2.66510948]), np.array([ 0.59030322, 30.55082687, -0.46183924, 4.39769959, 2.55446814]), np.array([ 0.53114467, 39.20861977, -17.93187819, 20.60397171, 2.27692367])])
+        ## LRF values for individual PMTs
+        LRF_PMTs_array = np.array([(self.eta(np.sqrt((x - PMT_position_x[j-1])**2 + (y - PMT_position_y[j-1])**2), *fitparameters[j-1])) for j in range(1,8)])
```

Comment:
The offset by one isn't needed as only the area from the top array is passed through.

Reply:
Right, it also worked like that, the shift in indices was only an ugly relic from the original code using pandas dataframes before being boiled down to the minimalist version it is now. Corrected it now and also accounted for `pmt_mask`.